src/adapter/src/config/frontend.rs

```diff
@@ -33,17 +35,48 @@ pub struct SystemParameterFrontend {
 impl SystemParameterFrontend {
     /// Construct a new [SystemParameterFrontend] instance.
     pub fn new(
+        env_id: EnvironmentId,
         ld_sdk_key: &str,
-        ld_user_key: &str,
         ld_key_map: BTreeMap<String, String>,
     ) -> Result<Self, anyhow::Error> {
         let ld_config = ld::ConfigBuilder::new(ld_sdk_key).build();
         let ld_client = ld::Client::build(ld_config)?;
-        let ld_user = ld::User::with_key(ld_user_key).build();
+        let ld_ctx = if env_id.cloud_provider() != "local" {
```

Comment:
String matching here is a little scary, because nothing will break if we rename `"local"`. Can you introduce an enum next to `EnvironmentId`?

```
/// Identifies a supported cloud provider.
enum CloudProvider {
   /// A local development environment rather than a cloud provider.
    Local,
   /// Amazon Web Services.
    Aws,
}

impl FromStr for CloudProvider {
    fn from_str(s: &str) -> CloudProvider {
        match s {
            "aws" => CloudProvider::Aws,
            "local" => CloudProvider::Local,
        }
    }
}
```

Reply:
[This is quite a lot of boilerplate code](https://github.com/MaterializeInc/materialize/pull/16678/commits/e6f0dfd52f9167c994cc1b281a748d8c69fa35a0#diff-4378b7fa6ab3cf0645ca66db0c08948910113c43e785c5a5fb9ea2e6de523905R672-R722) for introducing an enum of nullary variants and a mapping between that enum and `String`... surely there is a more compact way to do that with the right macro?